Mobileapps 2170

swuforesee opened the pull request:
*Changes:*
* Introduce uri for authentication and uri for service calls in the AcsApiClientConfig
* Always use the uri for authentication to load token (after 18.1 the endpoint for portal2 authentication will no longer by supported. This ensures all authentication goes to services edge endpoint)
* Remove old get token implementation
* Migrated acs-apiclient-ios project and OAuth_IOS project to Xamarin.iOS Unified API (as suggested by Xamarin) 
* Create unit test project for acs-apiclient_android project and acs-apiclient-ios project

@ahusseini could you please particularly take a look at the iOS projects since the migration could have made changes with consequences that I may not have realized? Please could you pass to Selwyn after your review?

@sleeke after your review, do you think its worthwhile to ask Shailesh to review the changes?

ahusseini commented on AndroidUnitTests/Tests.cs:
```diff
@@ -0,0 +1,57 @@
+namespace AndroidUnitTests
+{
+    [TestFixture]
+    public class Tests
```
:bowtie: These are technically integration tests, not unit tests

swuforesee commented:
*Changes:*
* Removed:
- acs-apiclient_android.csproj
- acs-apiclient-ios.csproj
- OAuth_IOS.csproj
- OAuth_Android.csproj

* Created Console Sample App
* Updated README
* Add gitignore for /bin and /obj folders in Console Sample App
